### PR TITLE
Add update task command and its parser

### LINKED ISSUE #50: Add basic implementation of update command
Basic implementation of update command to be added into the program

### LINKED ISSUE #67: Add parser for UpdateTaskCommand
Add parser for UpdateTaskCommand

### PR DESCRIPTION
**Pull request by ivorcmx**
Add UpdateTaskCommand to allow users to update description and/or deadline in the current tasklist
Add UpdateTaskCommandParser to allow parsing of the new command

Update AddressBookParser and ModelManager files accordingly

Fixes Issue #50 and Issue #67

### CODE REVIEW

**Comment by ivorcmx** [comment]
Tests for the new classes will be added in due to time to increase coverage

File: src/main/java/seedu/address/model/task/Task.java [hunk 1]
```diff
@@ -3,8 +3,9 @@
 import static java.util.Objects.requireNonNull;
 
 public class Task {
-    private final String description;
```
Comment (putaojuice) [inline src/main/java/seedu/address/model/task/Task.java:6]: I don't think it is a good idea to make it not immutable

**Review by putaojuice — COMMENTED** [review]
Imo i think it is better to delete the task and add a new task when "updating" a task instead of changing the description/deadline to keep the Task objects immutable.

File: src/main/java/seedu/address/model/task/Task.java (hunk 1 above)
Comment (ivorcmx) [reply src/main/java/seedu/address/model/task/Task.java:6]: Agreed. Updated the implementation to ensure immutability of the task. Great suggestion putao!

Comment (yumengtan) [reply src/main/java/seedu/address/model/task/Task.java:6]: LFG!! :fire::fire::fire:

Comment (ivorcmx) [reply src/main/java/seedu/address/model/task/Task.java:6]: 👍 👍 👍 👍 👍 👍 👍

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -183,4 +188,11 @@ public String findTask(String input) {
     public String viewTask() {
         return taskList.viewTask();
     }
+
+    //=========== Update Task ===============
+    @Override
+    public void updateTask(Task taskToUpdate, Task updatedTask, Integer taskId) {
```
Comment (putaojuice) [inline src/main/java/seedu/address/model/ModelManager.java]: parameter taskToUpdate never used?

File: src/main/java/seedu/address/logic/parser/UpdateTaskCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+     * Adapted from AddTaskCommandParser
+     *
+     * @param argMultimap Tokenized user input
+     * @throws ParseException Throw exception if prefix token does not exist
+     */
+    private void checkMinimumOnePrefix(ArgumentMultimap argMultimap) throws ParseException {
+        List<String> description = argMultimap.getAllValues(PREFIX_ADD_TASK_DESCRIPTION);
+        List<String> deadline = argMultimap.getAllValues(PREFIX_ADD_TASK_DEADLINE);
+
+        if (description.size() + deadline.size() < 1) {
```
Comment (putaojuice) [inline src/main/java/seedu/address/logic/parser/UpdateTaskCommandParser.java:90]: Seems to be doing the same job as `checkEmptyPrefix()` but ok hahaha

File: src/main/java/seedu/address/logic/commands/UpdateTaskCommand.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+     * Creates and returns a {@code Task} with the details of {@code taskToEdit}
+     * edited with {@code updateTaskDescriptor}.
+     */
+    private static Task createUpdatedTask(Task taskToUpdate, UpdateTaskDescriptor updateTaskDescriptor) {
+        assert taskToUpdate != null;
+
+        String updatedDescription = "";
+        String updatedDeadline = "";
+
+        if (updateTaskDescriptor.getDescription() == null) {
```
Comment (putaojuice) [inline src/main/java/seedu/address/logic/commands/UpdateTaskCommand.java:77]: not sure if it is a good idea to use `== null` to compare, but if the tests covered it with no issues it should be fine

**Review by putaojuice — COMMENTED** [review]
One of the parameter not used

File: src/main/java/seedu/address/model/ModelManager.java (hunk 2 above)
Comment (ivorcmx) [reply src/main/java/seedu/address/model/ModelManager.java]: Fixed

File: src/main/java/seedu/address/logic/parser/UpdateTaskCommandParser.java (hunk 3 above)
Comment (ivorcmx) [reply src/main/java/seedu/address/logic/parser/UpdateTaskCommandParser.java:90]: Yes. Adapted from addTaskCommandParser class. Included in the javadocs comment as well

File: src/main/java/seedu/address/model/ModelManager.java (hunk 2 above)
Comment (yumengtan) [reply src/main/java/seedu/address/model/ModelManager.java]: thumbs up man! 👍

**Review by putaojuice — APPROVED** [review]
LGTM

**Pull request merged**